From 12e9b977e4c7515ce90fecc62630be394fd7da62 Mon Sep 17 00:00:00 2001
From: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>
Date: Mon, 20 Dec 2021 19:54:39 +0000
Subject: [PATCH] Change UID of variable index in SMM

This patch fixes the os_indications setVariable() failure. The variable
index UID in SMM gateway which was 1 is changed in this patch. TFM has a
special usage for variable with UID 1, which makes it write once only.
This is not required for SMM variable index.

Upstream-Status: Pending [Not submitted to upstream yet]
Signed-off-by: Gowtham Suresh Kumar <gowtham.sureshkumar@arm.com>


---
 components/service/smm_variable/backend/uefi_variable_store.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/service/smm_variable/backend/uefi_variable_store.c b/components/service/smm_variable/backend/uefi_variable_store.c
index 6a90f46a..1bb869ae 100644
--- a/components/service/smm_variable/backend/uefi_variable_store.c
+++ b/components/service/smm_variable/backend/uefi_variable_store.c
@@ -67,7 +67,7 @@ static efi_status_t check_name_terminator(
 	size_t name_size);
 
 /* Private UID for storing the variable index */
-#define VARIABLE_INDEX_STORAGE_UID			(1)
+#define VARIABLE_INDEX_STORAGE_UID			(0x787)
 
 /* Default maximum variable size -
  * may be overridden using uefi_variable_store_set_storage_limits()
